pkg/scheduler/scheduler.go

```diff
@@ -448,6 +459,31 @@ func (s *Scheduler) getAssignments(log logr.Logger, wl *workload.Info, snap *cac
 	return fullAssignment, nil
 }
 
+func updateAssignmentForTAS(cq *cache.ClusterQueueSnapshot, wl *workload.Info, assignment flavorassigner.Assignment, targets []*preemption.Target) {
+	if features.Enabled(features.TopologyAwareScheduling) && assignment.RepresentativeMode() == flavorassigner.Preempt && wl.IsRequestingTAS() {
+		tasRequests := assignment.WorkloadsTopologyRequests(wl, cq)
+		var tasResult cache.TASAssignmentsResult
+		if len(targets) > 0 {
+			var targetWorkloads []*workload.Info
+			for _, target := range targets {
+				targetWorkloads = append(targetWorkloads, target.WorkloadInfo)
+			}
+			revertUsage := cq.SimulateUsageRemoval(targetWorkloads)
+			tasResult = cq.FindTopologyAssignmentsForWorkload(tasRequests, false)
+			revertUsage()
```

Comment:
```suggestion
			revertUsage := cq.SimulateUsageRemoval(targetWorkloads)
			defer revertUsage()
			tasResult = cq.FindTopologyAssignmentsForWorkload(tasRequests, false)
```

Should we use `defer` statement to guarantee it?

Reply:
Oh, this seems not to work. NVM

Reply:
Thank you!

/lgtm
/approve